inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1050,8 +1054,27 @@ void MKLDNNGraphOptimizer::FuseConvolutionSumAndConvolutionSumActivation(MKLDNNG
         if (!isSuitableParent1 && !isSuitableParent2)
             continue;
 
-        auto mergedConv = isSuitableParent1 ? parent1 : parent2;
-        auto peerNode = isSuitableParent1 ? parent2 : parent1;
+        std::shared_ptr<MKLDNNNode> mergedConv;
+        std::shared_ptr<MKLDNNNode> peerNode;
+
+        if (isSuitableParent1 && isSuitableParent2) {
+            // not merged operation (peerNode) has to be in low precision
+            const auto branch2Parent = graphNode->getParentEdgesAtPort(1)[0]->getParent();
+            const auto fused = branch2Parent->getFusedWith();
+            auto branch2Precision = fused.empty() ?
+                branch2Parent->getOriginalOutputPrecisionAtPort(0) :
+                fused[fused.size() - 1]->getOriginalOutputPrecisionAtPort(0);
+            const auto isBranch2Quantized = (branch2Precision == Precision::I8) || (branch2Precision == Precision::U8);
+            const auto parent1CanBeMerged = parent1->getChildEdges().size() == 1ul;
+
+            // if both branches are quantized, then parent1 is selected (result is not changed)
+            mergedConv = isBranch2Quantized && parent1CanBeMerged ? parent1 : parent2;
+            peerNode = isBranch2Quantized && parent1CanBeMerged ? parent2 : parent1;
+        } else {
+            mergedConv = isSuitableParent1 ? parent1 : parent2;
+            peerNode = isSuitableParent1 ? parent2 : parent1;
+        }
```

Comment:
maybe I missed something but code on 1078-1084 set `mergedConv = parent2; peerNode = parent1;` regardless of the `isBranch2Quantized ` value